RangeHTTPServer.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python
+'''
+Use this in the same way as Python's SimpleHTTPServer:
+
+  python -m RangeHTTPServer [port]
+
+The only difference from SimpleHTTPServer is that RangeHTTPServer supports
+'Range:' headers to load portions of files. This is helpful for doing local web
+development with genomic data files, which tend to be to large to load into the
+browser all at once.
+'''
+
+import os
+import re
+import sys
+from SimpleHTTPServer import SimpleHTTPRequestHandler
+import SimpleHTTPServer
+
+
+def copy_byte_range(infile, outfile, start, stop=None, bufsize=16*1024):
+    '''Like shutil.copyfileobj, but only copy a range of the streams.
+
+    Both start and stop are inclusive.
+    '''
+    infile.seek(start)
+    while 1:
+        to_read = min(bufsize, stop + 1 - infile.tell() if stop else bufsize)
+        buf = infile.read(to_read)
+        if not buf:
+            break
+        outfile.write(buf)
+
+
+BYTE_RANGE_RE = re.compile(r'bytes=(\d+)-(\d+)')
+def parse_byte_range(byte_range):
+    '''Returns the two numbers in 'bytes=123-456' or throws ValueError.'''
+    m = BYTE_RANGE_RE.match(byte_range)
+    if not m:
+        raise ValueError('Invalid byte range %s' % byte_range)
```

Comment:
I verified that Apache behaves in the way described in that SO post.

I added support for `bytes=0-`. Unfortunately, `SimpleHTTPServer` doesn't distinguish between a header with an empty value and a header that's not present. So I can't support plain-old `Range:`. In practice, I don't think this will be a big deal but feel free to file an issue.